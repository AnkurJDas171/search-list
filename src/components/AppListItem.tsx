import React from "react";
import { StyleSheet, View } from "react-native";

import { AppListItemProps } from "./types";
import AppTextContainer from "../UI/AppTextContainer";
import colors from "../assets/colors";

const AppListItem = ({
    data,
    containerColor,
    textColor = colors.BLACK
}: AppListItemProps): JSX.Element => {
    return (
        <View style={[styles.body, { backgroundColor: containerColor }]}>
            <AppTextContainer title={data.name} textColor={textColor} />
            <AppTextContainer title={data.rank} textColor={textColor} />
            <AppTextContainer title={data.bananas} textColor={textColor} />
            <AppTextContainer title={data.isSearched} textColor={textColor} />
        </View>
    )
}

const styles = StyleSheet.create({
    body: {
        alignItems: 'center',
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'center',
    }
})

export default AppListItem;
