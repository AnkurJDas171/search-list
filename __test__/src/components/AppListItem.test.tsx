import React from "react";
import { render } from "@testing-library/react-native";

import AppListItem from "../../../src/components/AppListItem";
import mockUsersdata from "../../__mock__/data/UsersData";

describe("List Item", () => {
    it("should render list item data", () => {
        const screen = render(
            <AppListItem
                containerColor={"#eee"}
                data={mockUsersdata[0]}
                textColor={"#000"}
            />
        )
        const text = screen.getByText(mockUsersdata[0].name);

        expect(screen.toJSON()).toMatchSnapshot();
        expect(text).toBeTruthy();
    })
})
