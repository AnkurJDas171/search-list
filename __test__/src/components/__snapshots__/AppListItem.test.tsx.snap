// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item should render list item data 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#eee",
      },
    ]
  }
>
  <View
    style={
      {
        "width": "25%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="RNE__LISTITEM__padView"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "backgroundColor": "transparent",
            "color": "#000",
            "fontSize": 22,
            "textAlign": "left",
          }
        }
        testID="app-text-container"
      >
        Chris Buckley
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "width": "25%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="RNE__LISTITEM__padView"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "backgroundColor": "transparent",
            "color": "#000",
            "fontSize": 22,
            "textAlign": "left",
          }
        }
        testID="app-text-container"
      >
        1
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "width": "25%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="RNE__LISTITEM__padView"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "backgroundColor": "transparent",
            "color": "#000",
            "fontSize": 22,
            "textAlign": "left",
          }
        }
        testID="app-text-container"
      >
        36750
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "width": "25%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="RNE__LISTITEM__padView"
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "backgroundColor": "transparent",
            "color": "#000",
            "fontSize": 22,
            "textAlign": "left",
          }
        }
        testID="app-text-container"
      >
        false
      </Text>
    </View>
  </View>
</View>
`;
